stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -123,23 +185,33 @@ enum class LockContext {
   OtherAsynchronous
 };
 
+// We can do this because a task can only add status records to itself. If we
+// ever change addStatusRecord to allow it to add status records to a different
+// task, we will need to revisit this.
 static std::memory_order getLoadOrdering(LockContext lockContext) {
   return lockContext != LockContext::OnTask
                           ? std::memory_order_acquire
                           : std::memory_order_relaxed;
 }
 
-/// Call the given function while holding the task status record lock.
+/// Makes sure to call the `fn` while holding the StatusRecordLock of the
+/// input task.
 ///
-/// The value in `status` will be updated with the current status value
-/// (ignoring the `TaskStatusLockRecord`) before calling the function,
-/// and the value there will be written back into the task status after
-/// calling the function.
-template <class Fn>
+/// If the client of withStatusRecordLock has already loaded the status of the
```

Comment:
Looks like the status parameter isn't optional, so this should be rephrased as a requirement rather than "if... may."

Reply:
Well it still is, there is still the variant that does this: 

```
/// A convenience version of the above for contexts that haven't already
/// done the load.
template <class Fn>
static bool withStatusRecordLock(AsyncTask *task,
                                 LockContext lockContext,
                                 Fn &&fn) {
  auto loadOrdering = getLoadOrdering(lockContext);
  ActiveTaskStatus status =
    task->_private()._status().load(loadOrdering);
  if (loadOrdering == std::memory_order_acquire)
    _swift_tsan_acquire(task);
  return withStatusRecordLock(task, lockContext, status, [&](ActiveTaskStatus taskStatus) {
    fn(taskStatus);
  });
}
```

Reply:
Ah, I gotcha. A brief mention of the existence of that variant might help reduce confusion in oblivious sorts like myself.